packages/application-shell/src/components/login-sso-callback/login-sso-callback.js

```diff
@@ -31,16 +28,20 @@ export class LoginSSOCallback extends React.PureComponent {
     const fragments = qs.parse(this.props.location.hash.substring(1));
     const idToken = fragments.id_token;
     const decodedIdToken = jwtDecode(idToken);
-    const nonce = storage.get(STORAGE_KEYS.NONCE);
-    storage.remove(STORAGE_KEYS.NONCE);
+    const sessionState = storage.get(
+      `${STORAGE_KEYS.NONCE}_${decodedIdToken.nonce}`,
+      { storage: window.sessionStorage }
```

Comment:
I'm not 100% happy with this...I'm not sure having a `storage` package is also really necessary.